plugins/discovery-ec2/src/main/java/org/elasticsearch/plugin/discovery/ec2/Ec2DiscoveryPlugin.java

```diff
@@ -123,4 +139,41 @@ public void onModule(DiscoveryModule discoveryModule) {
         // Register cloud node settings: cloud.node
         AwsEc2Service.AUTO_ATTRIBUTE_SETTING);
     }
+
+    /** Adds a node attribute for the ec2 availability zone. */
+    @Override
+    public Settings additionalSettings() {
+        if (AwsEc2Service.AUTO_ATTRIBUTE_SETTING.get(settings) == false) {
```

Comment:
Can you move all that code to AwsEc2ServiceImpl.getEc2Attributes instead?

Potentially we could unit test this method.